src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const result = [ ];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties' :
+        Object.assign(stateCopy, extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const value of keysToRemove) {
+          delete stateCopy[value];
+        }
+        break;
+
+      case 'clear' :
+        for (const key of Object.keys(stateCopy)) {
+          delete stateCopy[key];
+        }
```

Comment:
you can do like this, imho it's better to understand:
```suggestion
        for (const key in stateCopy) {
          delete stateCopy[key];
        }
```